tests/controllers/request.test.ts

```diff
@@ -0,0 +1,175 @@
+import { ObjectId } from 'mongodb';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import mongoose from 'mongoose';
+import Request from '../../src/models/request';
+
+import {
+  getAllRequests,
+  getRequest,
+  getRequestsOfUser,
+  getRequestedByUser,
+  createRequest,
+  updateRequest,
+  deleteRequest,
+} from '../../src/controllers/request';
+
+import { requestStatus } from '../../src/types';
+
+import {
+  mockBuyerUser,
+  mockSellerService,
+  mockBuyerRequest,
+  mockBuyerRequest2,
+  mockSellerUser,
+} from '../helpers/review';
+
+describe('Controller: Services', () => {
+  let mongoServer: MongoMemoryServer;
+
+  beforeAll(async () => {
+    mongoServer = await MongoMemoryServer.create();
+
+    await mongoose.connect(mongoServer.getUri());
+
+    await mockBuyerUser.save();
+    await mockSellerUser.save();
+    await mockSellerService.save();
+    await mockBuyerRequest.save();
+  });
+
+  afterAll(async () => {
+    if (mongoServer) {
+      await mongoose.connection.dropDatabase();
+      await mongoose.connection.close();
+      await mongoServer.stop();
+    }
+  });
+
+  it('Should be able to fetch all requests', async () => {
+    let getAllRequestsQuery = await getAllRequests();
+    expect(getAllRequestsQuery.status).toBe(true);
+    expect(getAllRequestsQuery.data).toBeInstanceOf(Array);
+    expect(getAllRequestsQuery.data?.length).toBe(1);
+
+    await mockBuyerRequest2.save();
+
+    getAllRequestsQuery = await getAllRequests();
+    expect(getAllRequestsQuery.status).toBe(true);
+    expect(getAllRequestsQuery.data).toBeInstanceOf(Array);
+    expect(getAllRequestsQuery.data?.length).toBe(2);
+    expect(getAllRequestsQuery.data?.at(0)?._id).toStrictEqual(
+      mockBuyerRequest._id
+    );
+    expect(getAllRequestsQuery.data?.at(1)?._id).toStrictEqual(
+      mockBuyerRequest2._id
+    );
+  });
+
+  it('Should be able to fetch a request by ID', async () => {
+    let getRequestQuery = await getRequest(new ObjectId().toString());
```

Comment:
There's a tiny tiny tiny smol chance of the random object ID actually matching a request but lite lol